zerver/middleware.py

```diff
@@ -306,14 +306,16 @@ def __call__(self, request: HttpRequest) -> HttpResponse:
             unset_request()
 
 
-def parse_client(request: HttpRequest) -> Tuple[str, Optional[str]]:
+@has_request_variables
+def parse_client(
+    request: HttpRequest,
+    req_client: Optional[str] = REQ("client", default=None, documentation_pending=True),
```

Comment:
I wasn't sure if this should be `documentation_pending` or `intentionally_undocumented`.

I'm not sure we would want to document this parameter in all endpoints. I think it could be documented in the [potential `HTTP headers` page](https://github.com/zulip/zulip/pull/20928) because it's connected to the parsing of the UserAgent...

But even so, we would want this marked as not documented for testing purposes, correct?

Reply:
Yeah, I think we want to cover this on the HTTP headers page, and  `intentionally_undocumented` with a comment that this is a common element to all API endpoints that we don't want to document on each endpoint's accepted parameters makes sense to me.